lib/libc/newlib/assert.c

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/__assert.h>
+
+/**
+ *
+ * @brief Handler of C standard library assert.
+ */
+void __assert_func(const char *file,
+		int line,
+		const char *func,
+		const char *failedexpr)
+{
+	__ASSERT_LOC(...);
+	__ASSERT_MSG_INFO("Assertion fail: (%s) at %s:%d",
+			failedexpr, file, line);
+	__ASSERT_POST_ACTION();
+}
```

Comment:
Instead of having this function in a separate file, given that it is unlikely we will ever add more assert-related stuff here, can we move it into `libc-hooks.c` as with the rest of the newlib hook functions?